unit_test/tstHertzJKR.hpp

```diff
@@ -0,0 +1,149 @@
+    //                  Discretization
+    // ====================================================
+    std::array<double, 3> low_corner = inputs["low_corner"];
+    std::array<double, 3> high_corner = inputs["high_corner"];
+    std::array<int, 3> num_cells = inputs["num_cells"];
+
+    // ====================================================
+    //            Custom particle creation
+    // ====================================================
+    const int num_particles = 2;
+    // Purposely using zero-init here.
+    Kokkos::View<double* [3], memory_space> position( "custom_position", 2 );
+    Kokkos::View<double*, memory_space> volume( "custom_volume", 2 );
+
+    Kokkos::parallel_for(
+        "create_particles", Kokkos::RangePolicy<exec_space>( 0, num_particles ),
+        KOKKOS_LAMBDA( const int p ) {
+            if ( p == 0 )
+                position( p, 0 ) = 5.01e-5;
+            else
+                position( p, 0 ) = -5.01e-5;
+            volume( p ) = vol;
+        } );
+
+    // ====================================================
+    //            Force model
+    // ====================================================
+    using model_type = CabanaPD::HertzianJKRModel;
+    // No search radius extension.
+    model_type contact_model( radius, radius_extend, nu, E, e, gamma );
+
+    // ====================================================
+    //                 Particle generation
+    // ====================================================
+    int halo_width = 1;
+    CabanaPD::Particles particles( memory_space{}, model_type{}, position,
+                                   volume, low_corner, high_corner, num_cells,
+                                   halo_width, exec_space{} );
+
+    // ====================================================
+    //            Custom particle initialization
+    // ====================================================
+    auto rho = particles.sliceDensity();
+    auto v = particles.sliceVelocity();
+    auto vo = particles.sliceVolume();
+
+    auto init_functor = KOKKOS_LAMBDA( const int p )
+    {
+        // Density
+        rho( p ) = rho0;
+        if ( p == 0 )
+            v( p, 0 ) = -1.0;
+        else
+            v( p, 0 ) = 1.0;
+    };
+    particles.updateParticles( exec_space{}, init_functor );
+
+    // Get initial total KE
+    // double ke_i = calculateKE( v, rho, vo );
+
+    // ====================================================
+    //  Simulation run
+    // ====================================================
+    CabanaPD::Solver solver( inputs, particles, contact_model );
+    solver.init();
+
+    Kokkos::View<double*> forces( "forces", solver.num_steps );
+
+    auto force = particles.sliceForce();
+
+    for ( int step = 1; step <= solver.num_steps; ++step )
+    {
+        solver.runStep( step );
+        forces( step ) = force( 0, 0 );
+    }
+
+    double min_po = getPulloffForce( force, vol );
+    EXPECT_EQ( min_po, 0.0 );
```

Comment:
This is a slightly dangerous test in that if nothing happened, it could still be zero.